### ISSUE TITLE
[BUG]: src.aihawk_easy_applier:_get_job_description:239 - Job description not found : Fails to apply using "Easy Apply" when using Linkedin Premium

### ISSUE THREAD

**Issue by craschnet** [feder-cr/Auto_Jobs_Applier_AIHawk#457]
### Describe the bug

Fails to apply for job due to trying to access easy button element when subscribed to LinkedIn Premium 

### Steps to reproduce
Subscribe to LinkedIn Premium (https://premium.linkedin.com/)
Installed via guide.
Main
Ran per instruction in virtual env as per guide:
(virtual) crasch@ub2204dt:~/Auto_Jobs_Applier_AIHawk$ python3 main.py 
Also the same error using "--resume <path to my resume>" option.
See Screen Recording: https://youtu.be/x3RuAm0zIaY
Attached app.log: [app.log](https://github.com/user-attachments/files/17212907/app.log)
No jobs are being applied to. 

```
2024-10-01 08:58:38.250 | DEBUG    | src.utils:scroll_slow:106 - Scrolled to final position: 0
2024-10-01 08:58:38.751 | DEBUG    | src.aihawk_easy_applier:_find_easy_apply_button:177 - Attempting search using find all 'Easy Apply' buttons using find_elements
2024-10-01 08:58:49.041 | WARNING  | src.aihawk_easy_applier:_find_easy_apply_button:191 - Button 1 found but not clickable: Message: 

2024-10-01 08:58:49.086 | DEBUG    | src.aihawk_easy_applier:_find_easy_apply_button:188 - Found 'Easy Apply' button 2, attempting to click
2024-10-01 08:58:49.091 | DEBUG    | src.aihawk_easy_applier:job_apply:118 - Retrieving job description
2024-10-01 08:58:49.092 | DEBUG    | src.aihawk_easy_applier:_get_job_description:223 - Getting job description
2024-10-01 08:58:49.121 | DEBUG    | src.aihawk_easy_applier:_get_job_description:232 - See more button not found, skipping
2024-10-01 08:58:49.148 | ERROR    | src.aihawk_easy_applier:_get_job_description:239 - Job description not found: Traceback (most recent call last):
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/src/aihawk_easy_applier.py", line 234, in _get_job_description
    description = self.driver.find_element(By.CLASS_NAME, 'jobs-description-content__text').text
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 831, in find_element
    return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":".jobs-description-content__text"}
  (Session info: chrome=129.0.6668.70)
Stacktrace:
#0 0x609a23bf910a <unknown>
#1 0x609a238df5e0 <unknown>
#2 0x609a2392ebe8 <unknown>
#3 0x609a2392ee81 <unknown>
#4 0x609a239758c4 <unknown>
#5 0x609a23953b4d <unknown>
#6 0x609a23972d7d <unknown>
#7 0x609a239538c3 <unknown>
#8 0x609a239216b3 <unknown>
#9 0x609a2392268e <unknown>
#10 0x609a23bc3b0b <unknown>
#11 0x609a23bc7a91 <unknown>
#12 0x609a23bb0305 <unknown>
#13 0x609a23bc8612 <unknown>
#14 0x609a23b9546f <unknown>
#15 0x609a23be8008 <unknown>
#16 0x609a23be81d3 <unknown>
#17 0x609a23bf7f5c <unknown>
#18 0x7bb56be94ac3 <unknown>


2024-10-01 08:58:49.148 | ERROR    | src.aihawk_easy_applier:job_apply:143 - Failed to apply to job: Job(title='B2C Security Engineer', company='K&K Global Talent Solutions INC.', location='United States (Remote)', link='https://www.linkedin.com/jobs/view/4034314589/', apply_method='Easy Apply', description='', summarize_job_description='', pdf_path='', recruiter_link=''), error: Traceback (most recent call last):
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/src/aihawk_easy_applier.py", line 234, in _get_job_description
    description = self.driver.find_element(By.CLASS_NAME, 'jobs-description-content__text').text
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 831, in find_element
    return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":".jobs-description-content__text"}
  (Session info: chrome=129.0.6668.70)
Stacktrace:
#0 0x609a23bf910a <unknown>
#1 0x609a238df5e0 <unknown>
#2 0x609a2392ebe8 <unknown>
#3 0x609a2392ee81 <unknown>
#4 0x609a239758c4 <unknown>
#5 0x609a23953b4d <unknown>
#6 0x609a23972d7d <unknown>
#7 0x609a239538c3 <unknown>
#8 0x609a239216b3 <unknown>
#9 0x609a2392268e <unknown>
#10 0x609a23bc3b0b <unknown>
#11 0x609a23bc7a91 <unknown>
#12 0x609a23bb0305 <unknown>
#13 0x609a23bc8612 <unknown>
#14 0x609a23b9546f <unknown>
#15 0x609a23be8008 <unknown>
#16 0x609a23be81d3 <unknown>
#17 0x609a23bf7f5c <unknown>
#18 0x7bb56be94ac3 <unknown>


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/src/aihawk_easy_applier.py", line 119, in job_apply
    job_description = self._get_job_description()
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/src/aihawk_easy_applier.py", line 240, in _get_job_description
    raise Exception(f"Job description not found: \nTraceback:\n{tb_str}")
Exception: Job description not found: 
Traceback:
Traceback (most recent call last):
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/src/aihawk_easy_applier.py", line 234, in _get_job_description
    description = self.driver.find_element(By.CLASS_NAME, 'jobs-description-content__text').text
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 831, in find_element
    return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":".jobs-description-content__text"}
  (Session info: chrome=129.0.6668.70)
Stacktrace:
#0 0x609a23bf910a <unknown>
#1 0x609a238df5e0 <unknown>
#2 0x609a2392ebe8 <unknown>
#3 0x609a2392ee81 <unknown>
#4 0x609a239758c4 <unknown>
#5 0x609a23953b4d <unknown>
#6 0x609a23972d7d <unknown>
#7 0x609a239538c3 <unknown>
#8 0x609a239216b3 <unknown>
#9 0x609a2392268e <unknown>
#10 0x609a23bc3b0b <unknown>
#11 0x609a23bc7a91 <unknown>
#12 0x609a23bb0305 <unknown>
#13 0x609a23bc8612 <unknown>
#14 0x609a23b9546f <unknown>
#15 0x609a23be8008 <unknown>
#16 0x609a23be81d3 <unknown>
#17 0x609a23bf7f5c <unknown>
#18 0x7bb56be94ac3 <unknown>



2024-10-01 08:58:49.149 | DEBUG    | src.aihawk_easy_applier:job_apply:145 - Discarding application due to failure
2024-10-01 08:58:49.149 | DEBUG    | src.aihawk_easy_applier:_discard_application:316 - Discarding application
2024-10-01 08:58:49.177 | WARNING  | src.aihawk_easy_applier:_discard_application:323 - Failed to discard application: Message: no such element: Unable to locate element: {"method":"css selector","selector":".artdeco-modal__dismiss"}
  (Session info: chrome=129.0.6668.70)
Stacktrace:
#0 0x609a23bf910a <unknown>
#1 0x609a238df5e0 <unknown>
#2 0x609a2392ebe8 <unknown>
#3 0x609a2392ee81 <unknown>
#4 0x609a239758c4 <unknown>
#5 0x609a23953b4d <unknown>
#6 0x609a23972d7d <unknown>
#7 0x609a239538c3 <unknown>
#8 0x609a239216b3 <unknown>
#9 0x609a2392268e <unknown>
#10 0x609a23bc3b0b <unknown>
#11 0x609a23bc7a91 <unknown>
#12 0x609a23bb0305 <unknown>
#13 0x609a23bc8612 <unknown>
#14 0x609a23b9546f <unknown>
#15 0x609a23be8008 <unknown>
#16 0x609a23be81d3 <unknown>
#17 0x609a23bf7f5c <unknown>
#18 0x7bb56be94ac3 <unknown>

2024-10-01 08:58:49.177 | ERROR    | src.aihawk_job_manager:apply_jobs:308 - Failed to apply for B2C Security Engineer at K&K Global Talent Solutions INC.: Failed to apply to job! Original exception:
Traceback:
Traceback (most recent call last):
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/src/aihawk_easy_applier.py", line 234, in _get_job_description
    description = self.driver.find_element(By.CLASS_NAME, 'jobs-description-content__text').text
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 831, in find_element
    return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":".jobs-description-content__text"}
  (Session info: chrome=129.0.6668.70)
Stacktrace:
#0 0x609a23bf910a <unknown>
#1 0x609a238df5e0 <unknown>
#2 0x609a2392ebe8 <unknown>
#3 0x609a2392ee81 <unknown>
#4 0x609a239758c4 <unknown>
#5 0x609a23953b4d <unknown>
#6 0x609a23972d7d <unknown>
#7 0x609a239538c3 <unknown>
#8 0x609a239216b3 <unknown>
#9 0x609a2392268e <unknown>
#10 0x609a23bc3b0b <unknown>
#11 0x609a23bc7a91 <unknown>
#12 0x609a23bb0305 <unknown>
#13 0x609a23bc8612 <unknown>
#14 0x609a23b9546f <unknown>
#15 0x609a23be8008 <unknown>
#16 0x609a23be81d3 <unknown>
#17 0x609a23bf7f5c <unknown>
#18 0x7bb56be94ac3 <unknown>


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/src/aihawk_easy_applier.py", line 119, in job_apply
    job_description = self._get_job_description()
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/src/aihawk_easy_applier.py", line 240, in _get_job_description
    raise Exception(f"Job description not found: \nTraceback:\n{tb_str}")
Exception: Job description not found: 
Traceback:
Traceback (most recent call last):
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/src/aihawk_easy_applier.py", line 234, in _get_job_description
    description = self.driver.find_element(By.CLASS_NAME, 'jobs-description-content__text').text
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 831, in find_element
    return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "/home/crasch/Auto_Jobs_Applier_AIHawk/virtual/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":".jobs-description-content__text"}
  (Session info: chrome=129.0.6668.70)
Stacktrace:
#0 0x609a23bf910a <unknown>
#1 0x609a238df5e0 <unknown>
#2 0x609a2392ebe8 <unknown>
#3 0x609a2392ee81 <unknown>
#4 0x609a239758c4 <unknown>
#5 0x609a23953b4d <unknown>
#6 0x609a23972d7d <unknown>
#7 0x609a239538c3 <unknown>
#8 0x609a239216b3 <unknown>
#9 0x609a2392268e <unknown>
#10 0x609a23bc3b0b <unknown>
#11 0x609a23bc7a91 <unknown>
#12 0x609a23bb0305 <unknown>
#13 0x609a23bc8612 <unknown>
#14 0x609a23b9546f <unknown>
#15 0x609a23be8008 <unknown>
#16 0x609a23be81d3 <unknown>
#17 0x609a23bf7f5c <unknown>
#18 0x7bb56be94ac3 <unknown>



2024-10-01 08:58:49.177 | DEBUG    | src.aihawk_job_manager:write_to_file:313 - Writing job application result to file: failed
2024-10-01 08:58:49.178 | DEBUG    | src.aihawk_job_manager:write_to_file:340 - Job data appended to existing file: failed

```

### Expected behavior

To apply for job using easy apply button

### Actual behavior

Fails due to not using the easy apply button.

### Branch

other

### Branch name

Main

### Python version

3.10.12

### LLM Used

ChatGPT 

### Model used

gpt-4o

### Additional context

- Tired on both main & v4 branches. 
- Also tested on Win10 and Ubuntu Desktop 22.04.

Affecting at least another users:
https://github.com/feder-cr/Auto_Jobs_Applier_AIHawk/issues/438

**Comment by naganikshith04** [comment]
Solution?

**Comment by craschnet** [comment]
Looks like an issue with LinkedIn Premium element names.

Standard LinkedIn:
![461879731_452856414471807_3623469495059881052_n](https://github.com/user-attachments/assets/efc3ca07-cf01-49c8-ae76-4b176215328b)

Linkedin Premium:
![image](https://github.com/user-attachments/assets/22369dad-0f90-4ed0-8a73-81a19153fc95)

This means, that there are multiple elements that are different when subscribing to Linkedin Premium

**Comment by queukat** [comment]
fixed in https://github.com/feder-cr/Auto_Jobs_Applier_AIHawk/pull/473

**Comment by queukat** [comment]
@feder-cr can be closed

**Comment by craschnet** [comment]
> @feder-cr can be closed

Thank you for looking into this!

**Comment by craschnet** [comment]
Still having the issue:
Tried the newer v4 branch: 
[error.txt](https://github.com/user-attachments/files/17417067/error.txt)

**Comment by xTaylorFerg** [comment]
Fixed this bug in my PR. Will also need to be pulled into the v4 branch.

**Comment by xTaylorFerg** [comment]
@craschnet can you close this issue? it was fixed with the following commit:
https://github.com/feder-cr/Auto_Jobs_Applier_AIHawk/pull/562

**Comment by craschnet** [comment]
> @craschnet can you close this issue? it was fixed with the following PR: #562
Will do Thx
